uavcan/driver.py

```diff
@@ -98,7 +108,10 @@ def close(self):
             libc.close(self.fd)
 
     def get_socket(ifname):
+        on  = ctypes.c_int(1)
         socket_fd = libc.socket(AF_CAN, socket.SOCK_RAW, CAN_RAW)
+        error = libc.setsockopt(socket_fd , SOL_SOCKET, socket.SO_TIMESTAMP, ctypes.byref(on), ctypes.sizeof(on))
+        error = libc.setsockopt(socket_fd, SOL_CAN_RAW, CAN_RAW_RECV_OWN_MSGS, ctypes.byref(on), ctypes.sizeof(on))
```

Comment:
Enabling reception of own messages causes the recv() call to return the same messages we're sending. While it is useful to implement TX timestamping, it requires some logic to prevent these loop-back frames from propagating to the higher levels of the stack. In libuavcan this logic is implemented by means of checking [certain flags](https://github.com/UAVCAN/uavcan/blob/master/libuavcan_drivers/linux/include/uavcan_linux/socketcan.hpp#L233).